core/services/pipeline/task.jsonparse.go

```diff
@@ -42,12 +43,19 @@ func (t *JSONParseTask) Run(_ context.Context, _ logger.Logger, vars Vars, input
 		return Result{Error: errors.Wrap(err, "task inputs")}, runInfo
 	}
 
+	var sep StringParam
+	err = errors.Wrap(ResolveParam(&sep, From(t.Separator)), "separator")
```

Comment:
If we decided (did we?) that allowed separators are `.` and `,`, then should not we check `sep` is one of these and return error otherwise?

Reply:
I believe this needs to be fully general in order to work around any possible field names. Even more than one character in some cases.